pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -145,6 +146,11 @@ public HelixServerStarter(String helixClusterName, String zkServer, Configuratio
 
     LOGGER.info("Connecting Helix manager");
     _helixManager.connect();
+    // Overwrite the server netty host and port.
+    if (_serverConf.getBoolean(CONFIG_OF_USE_LOGICAL_INSTANCE_ID, false)) {
```

Comment:
You could also simply look up if the server config has instance id set ?

Reply:
Similar consideration as above.